src/poetry/factory.py

```diff
@@ -205,23 +212,54 @@ def create_legacy_repository(
         )
 
     @classmethod
-    def create_pyproject_from_package(cls, package: ProjectPackage, path: Path) -> None:
+    def create_pyproject_from_package(
+        cls, package: Package, path: Path | None = None
+    ) -> TOMLDocument:
         import tomlkit
 
-        from poetry.layouts.layout import POETRY_DEFAULT
+        pyproject: dict[str, Any] = tomlkit.document()
+
+        tool_table = tomlkit.table()
+        tool_table._is_super_table = True
+        pyproject["tool"] = tool_table
 
-        pyproject: dict[str, Any] = tomlkit.loads(POETRY_DEFAULT)
-        content = pyproject["tool"]["poetry"]
+        content: dict[str, Any] = tomlkit.table()
+        pyproject["tool"]["poetry"] = content
 
         content["name"] = package.name
         content["version"] = package.version.text
         content["description"] = package.description
         content["authors"] = package.authors
-
-        dependency_section = content["dependencies"]
+        content["license"] = package.license.id if package.license else ""
+        content["classifiers"] = package.classifiers
+
+        for key, attr in {
+            ("documentation", "documentation_url"),
+            ("repository", "repository_url"),
+            ("homepage", "homepage"),
+            ("maintainers", "maintainers"),
+            ("keywords", "keywords"),
+        }:
+            if getattr(package, attr, None):
+                content[key] = getattr(package, attr)
+
+        content["readme"] = []
+
+        for readme in package.readmes:
+            try:
+                # since packages resolve readme paths, un-resolve it
+                readme_posix_path = readme.relative_to(package.root_dir).as_posix()
+            except ValueError:
+                # we use a try/except since `is_relative_to` is not available in
+                # Python < 3.9
+                readme_posix_path = readme.as_posix()
```

Comment:
That's true, but `is_relative_to` is not used here, only `relative_to`, which is available. It's even used in a test that did not fail. 😉

However, at least according to the type hints `package.root_dir` can be None. Thus, that should be handled somehow.

Reply:
Obviously :) it's there because `if path.is_relative_to()` seemed better and I had to replace that in a previous commit. And don't want people repeating my eagerness ;)